tests/std/tests/LWG2597_complex_branch_cut/test.cpp

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cmath>
+#include <complex>
+#include <limits>
+
+using namespace std;
+
+template <typename T>
+void check(T x, T y) {
+    constexpr T tol = 2 * numeric_limits<T>::epsilon();
+
+    if (isfinite(x) && isfinite(y)) {
+        assert(signbit(x) == signbit(y));
+        assert(abs(x - y) <= tol * fmax(abs(x), abs(y)));
+    } else if (isnan(x) || isnan(y)) {
+        assert(isnan(x) == isnan(y));
+    } else {
+        assert(x == y);
+    }
+}
+
+template <typename T>
+void check_part(const complex<T>& x, const complex<T>& y) {
+    check(x.real(), y.real());
+    check(x.imag(), y.imag());
+}
+
+template <typename T>
+void check_norm(const complex<T>& x, const complex<T>& y) {
+    constexpr T tol = 4 * numeric_limits<T>::epsilon();
+
+    if (isfinite(x.real()) && isfinite(x.imag()) && isfinite(y.real()) && isfinite(y.imag())) {
+        assert(abs(x - y) <= tol * fmax(abs(x), abs(y)));
+    } else {
+        const bool x_is_inf = isinf(x.real()) || isinf(x.imag());
+        const bool y_is_inf = isinf(y.real()) || isinf(y.imag());
+        assert(x_is_inf == y_is_inf);
+    }
+}
+
+template <typename T>
+void test() {
+    using CT = complex<T>;
+
+    constexpr T zero     = T{0};
+    constexpr T tiny     = numeric_limits<T>::denorm_min();
+    constexpr T log10_e  = T{0.43429448190325182765};
+    constexpr T half     = T{0.5};
+    constexpr T one_down = T{1} - numeric_limits<T>::epsilon() / T{numeric_limits<T>::radix};
+    constexpr T one      = T{1};
+    constexpr T one_up   = T{1} + numeric_limits<T>::epsilon();
+    constexpr T pi_2     = T{1.5707963267948966192};
+    constexpr T two      = T{2};
+    constexpr T pi       = T{3.1415926535897932385};

```

Comment:
For consistency with `<numbers>`:

```suggestion
    constexpr T pi       = T{3.141592653589793};
```